### PR TITLE
feat: show warning before blocking followed channel

### PR DESCRIPTION
**Pull request by Nerixyz**
When blocking a followed channel, a warning is now shown about unfollowing that channel as a result of blocking them.

Closes https://github.com/Chatterino/chatterino2/discussions/5565.
Closes https://github.com/Chatterino/chatterino2/issues/1886.

### CODE REVIEW

**Comment by 8thony** [comment]
Probably #1886 will be resolved as well

File: src/widgets/dialogs/UserInfoPopup.cpp [hunk 1]
```diff
@@ -663,15 +663,74 @@ void UserInfoPopup::installEvents()
...
+                                QString("User %1 couldn't be blocked, an "
+                                        "unknown error occurred!")
                                     .arg(this->userName_));
                             reenableBlockCheckbox();
                         });
-                }
-                break;
+                };
+                getHelix()->getFollowedChannel(
+                    getApp()->getAccounts()->twitch.getCurrent()->getUserId(),
```
Comment (pajlada) [inline src/widgets/dialogs/UserInfoPopup.cpp:673]: Let's have the messagebox always show up, regardless of follow state

File: src/providers/twitch/api/Helix.cpp [hunk 2]
```diff
@@ -3138,7 +3138,7 @@ void Helix::getUserEmotes(
 }
 
 void Helix::getFollowedChannel(
-    QString userID, QString broadcasterID,
+    QString userID, QString broadcasterID, const QObject *caller,
```
Comment (pajlada) [inline src/providers/twitch/api/Helix.cpp:3141]: this change can stay

File: src/widgets/dialogs/UserInfoPopup.cpp [hunk 3]
```diff
@@ -663,15 +663,74 @@ void UserInfoPopup::installEvents()
...
+                                     followStatus->followedAt.toString()),
+                            QMessageBox::Yes | QMessageBox::Cancel,
+                            QMessageBox::Cancel);
+                        if (wasPinned)
+                        {
+                            this->togglePinned();
+                        }
+                        if (btn != QMessageBox::Yes)
+                        {
+                            reenableBlockCheckbox();
```
Comment (pajlada) [inline src/widgets/dialogs/UserInfoPopup.cpp:698]: Pressing "Cancel" will leave the checkbox toggled

**Review by pajlada — CHANGES_REQUESTED** [review]

**Review by pajlada — APPROVED** [review]

**Pull request merged**